core/variant.cpp

```diff
@@ -601,6 +601,31 @@ bool Variant::can_convert_strict(Variant::Type p_type_from, Variant::Type p_type
 	return false;
 }
 
+bool Variant::can_convert(Type p_type_to) const {
```

Comment:
The additional logic in `can_convert(Type p_type_to)` requires an instance of `Variant` which wouldn't be available from the static version of the method. Notice that line 609 of `core/variant.cpp` must call `operator String()` in order to fetch the value of the string that is attempting to be converted. The only solution I can see would be refactoring all parts of the code that call the static version of the method so they instead call the instance version of the method and move all the logic to the instance method.

Reply:
In my opinion, this approach is unacceptable. Please either choose a different name for the method, or make the behavior consistent for static and non-static versions, or replace `can_convert()` with in place checks or maybe even use `can_convert_strict()` (can be breaking).

Also, this requires research for compatibility with 3.6 and consistency with 4.x. If it has the same bug, then we should fix it in 4.x first. It is not desirable to have differences with 4.x in such methods without a good reason.

Reply:
Making the behavior consistent would require adding a third parameter to the static version of `Variant::can_convert` which accepts an instance of a `Variant`. This wouldn't work since in cases where there is no instance and would require a change in the function signature. It also would defeat the purpose of having the instance version of the method.

In place checks would be messy because they would require the type to be checked in order to know if the conversion is from a string to a numeric type beforehand. Which in turn would require an additional if block in every part of the code where `Variant::can_convert` is called.

Using `can_convert_strict` instead wouldn't make much sense and would still require a change in the function signature.

So the only option left is renaming the function.

This bug will need to be checked for in 4.x and fixed there if present. Until that is handled and compatibility with 3.6 is researched, there is no path forward for fixing this bug.